src/clients/java/src/client.zig

```diff
@@ -93,415 +378,305 @@ const ReflectionHelper = struct {
-            packet.user_data = global_ref;
-            packet.data = send_buffer.ptr;
-            packet.data_size = @intCast(u32, send_buffer.len);
-            packet.next = null;
-            packet.status = .ok;
-            tb.submit(context.client, packet);
-        } else {
-            assert(acquire_status != .ok);
-        }
-
-        return acquire_status;
     }
 
-    /// Completion callback.
-    fn on_completion(
-        context_ptr: usize,
-        client: tb.tb_client_t,
-        packet: *tb.tb_packet_t,
-        result_ptr: ?[*]const u8,
-        result_len: u32,
-    ) callconv(.C) void {
-        _ = client;
-
-        var context = @intToPtr(*JNIContext, context_ptr);
-
-        var env = context.jvm.attachCurrentThreadAsDaemon() catch |err| {
-            log.err("Unexpected error attaching the native thread as daemon {}", .{err});
-            @panic("JNI: Unexpected error attaching the native thread as daemon");
-        };
-
-        // Retrieves the request instance, and drops the GC reference.
-        assert(packet.user_data != null);
-        var request_obj = @ptrCast(jui.jobject, packet.user_data);
-        defer env.deleteReference(.global, request_obj);
-
-        const packet_operation = packet.operation;
-        const packet_status = packet.status;
-        if (result_len > 0) {
-            switch (packet_status) {
-                .ok => if (result_ptr) |ptr| {
-                    // Copying the reply before releasing the packet.
-                    ReflectionHelper.set_reply_buffer(
-                        env,
-                        request_obj,
-                        ptr[0..@intCast(usize, result_len)],
-                    );
-                },
-                else => {},
-            }
-        }
-        context.release_packet(packet);
-
-        ReflectionHelper.end_request(
+    pub inline fn find_field(
+        env: *jni.JNIEnv,
+        class: jni.JClass,
+        comptime name: [:0]const u8,
+        comptime signature: [:0]const u8,
+    ) jni.JFieldID {
+        return env.get_field_id(class, name.ptr, signature.ptr) orelse
+            vm_panic(
             env,
-            request_obj,
-            packet_operation,
-            packet_status,
+            "Field could not be found {s} {s}",
+            .{ name, signature },
         );
     }
-};
-
-/// Export function using the JNI calling convention.
-const Exports = struct {
-    pub fn on_load(vm: *jui.JavaVM) callconv(.C) jui.jint {
-        return NativeClient.on_load(vm);
-    }
-
-    pub fn on_unload(vm: *jui.JavaVM) callconv(.C) void {
-        NativeClient.on_unload(vm);
-    }
 
-    pub fn client_init(
-        env: *jui.JNIEnv,
-        class: jui.jclass,
-        cluster_id: jui.jint,
-        addresses: jui.jstring,
-        max_concurrency: jui.jint,
-    ) callconv(.C) jui.jlong {
-        _ = class;
-        var context = NativeClient.client_init(
-            false,
+    pub inline fn find_method(
```

Comment:
Can these find_  methods also assert that the result is not-null?

Reply:
It's done by the `orelse` clause:

https://github.com/tigerbeetle/tigerbeetle/blob/120e6a62ea6320d1f23b0c3b89e701693140ca57/src/clients/java/src/client.zig#L654-L659

Reply:
:face-palm: :sweat_smile: